freqtrade/strategy/interface.py

```diff
@@ -328,17 +328,24 @@ def get_signal(self, pair: str, interval: str, dataframe: DataFrame) -> Tuple[bo
         latest_date = dataframe['date'].max()
         latest = dataframe.loc[dataframe['date'] == latest_date].iloc[-1]
 
-        # Check if dataframe is out of date
         interval_minutes = timeframe_to_minutes(interval)
+
+        # Check if dataframe is out of date
         offset = self.config.get('exchange', {}).get('outdated_offset', 5)
         if latest_date < (arrow.utcnow().shift(minutes=-(interval_minutes * 2 + offset))):
             logger.warning(
                 'Outdated history for pair %s. Last tick is %s minutes old',
                 pair,
-                (arrow.utcnow() - latest_date).seconds // 60
+                int((arrow.utcnow() - latest_date).total_seconds() // 60)
             )
             return False, False
 
+        # Check if dataframe has new candle
+        if (arrow.utcnow() - latest_date).total_seconds() // 60 >= interval_minutes:
+            logger.warning('Old candle for pair %s. Last candle is %s minutes old',
+                           pair, int((arrow.utcnow() - latest_date).total_seconds() // 60))
+            return False, False
+
```

Comment:
In principle, this PR adds this new check only.

I did not get yet, what situation (which can happen in the user custom code) should it protect from...

Reply:
well and this check is not really new - but the same (but with twice the candle lenght) exists a little above or below this check.

The initial problem was that lock_pair locks until the candle is finished (so until 10:00 - but a new candle may only arrive at 10:00:05 - so we got 5s where freqtrade would buy into the "old" signal - even though no new candle arrived.

As explained above - to me this is the wrong approach to fixing the problem it tries to address, but instead the problem should be addressed within `is_pair_locked` (but all of this is above - although it can be a long read ...).